### PR TITLE
Touchpad-friently 3d navigation

### PR DESCRIPTION
**Pull request by buzzhuzz**
Implement FreeCAD-inspired touchpad navigation mode.

This mode makes 3d model inspection easier for devices with touchpad as the only pointer device.

Controls:
- Shift+move: panning
- Alt+move: rotation

### CODE REVIEW

**Comment by SoftFever** [comment]
@buzzhuzz 
Thank you for the PR, this sounding interesting.
Do you mind to explain a bit more how it works(screen recording may helps to demonstrate the feature)

File: src/slic3r/GUI/GLCanvas3D.cpp [hunk 1]
```diff
@@ -4452,6 +4456,24 @@ void GLCanvas3D::on_set_focus(wxFocusEvent& evt)
...
+    if (wxGetApp().app_config->get("camera_navigation_style") == "1") {
+        return evt.Moving() && evt.AltDown() && !evt.ShiftDown();
+    } else {
+        return evt.Dragging() && evt.LeftIsDown();
+    }
+}
+
+bool GLCanvas3D::is_camera_pan(const wxMouseEvent& evt) const
+{
+    if (wxGetApp().app_config->get("camera_navigation_style") == "1") {
```
Comment (SoftFever) [inline src/slic3r/GUI/GLCanvas3D.cpp:4470]: Same

**Comment by SoftFever** [comment]
Works like a charm ;)

**Comment by buzzhuzz** [comment]
I made a screen cap which demonstrates this feature in action.

Default navigation mode:

https://github.com/SoftFever/OrcaSlicer/assets/46728448/5001aec9-815b-4645-b618-272631c726e6


"Touchpad" navigation mode:

https://github.com/SoftFever/OrcaSlicer/assets/46728448/9cfffdb1-853b-4ac8-9a14-48a9f61f2f6b


This navigation modes splits scene control over two hands making scene control easier for devices like touchpad or some trackballs.

File: src/slic3r/GUI/GLCanvas3D.cpp [hunk 2]
```diff
@@ -4452,6 +4456,24 @@ void GLCanvas3D::on_set_focus(wxFocusEvent& evt)
     m_tooltip_enabled = true;
 }
 
+bool GLCanvas3D::is_camera_rotate(const wxMouseEvent& evt) const
+{
+    if (wxGetApp().app_config->get("camera_navigation_style") == "1") {
```
[the remark this answers is not in the document]
Comment (buzzhuzz) [reply src/slic3r/GUI/GLCanvas3D.cpp]: Updated that code.

Now this setting is cached on canvas focus. 

I've also added forced focus for the canvas on preferences window close.  This helps to solve 2 things:
1. Force update setting value (which also could be applied to "zoom_to_mouse", "reverse_mouse_wheel_zoom" and "use_free_camera")
2. User could immediately use 3D view hotkeys without clicking on a view first.

**Review by SoftFever — APPROVED** [review]
Looks good.
Thank you!

**Pull request merged**